app/src/main/java/com/google/samples/apps/sunflower/compose/garden/GardenScreen.kt

```diff
@@ -0,0 +1,263 @@
+import java.util.*
+
+@Composable
+fun GardenScreen(
+    viewModel: GardenPlantingListViewModel = viewModel(),
+    onAddPlantClick: () -> Unit,
+    onPlantClick: (PlantAndGardenPlantings) -> Unit
+) {
+    val gardenPlants = viewModel.plantAndGardenPlantings.observeAsState().value
+    GardenScreen(
+        gardenPlants = gardenPlants,
+        onAddPlantClick = onAddPlantClick,
+        onPlantClick = onPlantClick
+    )
+}
+
+@Composable
+private fun GardenScreen(
+    gardenPlants: List<PlantAndGardenPlantings>?,
+    onAddPlantClick: () -> Unit = {},
+    onPlantClick: (PlantAndGardenPlantings) -> Unit = {}
+) {
+    if (gardenPlants == null || gardenPlants.isEmpty()) {
+        EmptyGarden(onAddPlantClick)
+    } else {
+        GardenList(gardenPlants, onPlantClick = onPlantClick)
+    }
+
+    // When gardenPlants is not null, the content should be fully drawn
+    ReportDrawnWhen {
+        gardenPlants != null
+    }
+}
+
+@Composable
+private fun GardenList(
+    gardenPlants: List<PlantAndGardenPlantings>,
+    onPlantClick: (PlantAndGardenPlantings) -> Unit
+) {
+    LazyVerticalGrid(
+        columns = GridCells.Fixed(2),
+        contentPadding = PaddingValues(
+            horizontal = dimensionResource(id = R.dimen.card_side_margin),
+            vertical = dimensionResource(id = R.dimen.margin_normal)
+        )
+    ) {
+        items(count = gardenPlants.size) { index ->
+            val plant = gardenPlants[index]
+            GardenListItem(plant = plant, onPlantClick = onPlantClick)
+        }
+    }
+}
+
+@OptIn(
+    ExperimentalMaterialApi::class, ExperimentalGlideComposeApi::class,
+    ExperimentalComposeUiApi::class
+)
+@Composable
+private fun GardenListItem(
+    plant: PlantAndGardenPlantings,
+    onPlantClick: (PlantAndGardenPlantings) -> Unit
+) {
+    val vm = PlantAndGardenPlantingsViewModel(plant)
+    Card(
+        onClick = { onPlantClick(plant) },
+        modifier = Modifier.padding(
+            start = dimensionResource(id = R.dimen.card_side_margin),
+            end = dimensionResource(id = R.dimen.card_side_margin),
+            bottom = dimensionResource(id = R.dimen.card_bottom_margin)
+        ),
+        elevation = dimensionResource(id = R.dimen.card_elevation),
+        shape = MaterialTheme.shapes.card,
+    ) {
+        Column(Modifier.fillMaxWidth()) {
+            GlideImage(
+                model = vm.imageUrl,
+                contentDescription = plant.plant.description,
+                Modifier
+                    .fillMaxWidth()
+                    .height(dimensionResource(id = R.dimen.plant_item_image_height)),
+                contentScale = ContentScale.Crop,
+            )
+
+            // Plant name
+            Text(
+                text = vm.plantName,
+                Modifier
+                    .padding(
+                        top = dimensionResource(id = R.dimen.margin_normal),
```

Comment:
would it be worth to extract the dimension into a variable to prevent parsing it multiple times?